persist/fs/msgpack/decoder.go

```diff
@@ -215,17 +223,36 @@ func (dec *Decoder) DecodeLogMetadata() (schema.LogMetadata, error) {
 }
 
 func (dec *Decoder) decodeIndexInfo() schema.IndexInfo {
-	numFieldsToSkip, ok := dec.checkNumFieldsFor(indexInfoType)
+	numFieldsToSkip, ok := dec.checkNumFieldsFor(indexInfoType, true)
 	if !ok {
 		return emptyIndexInfo
 	}
+
+	// -numFieldsAddedToIndexInfoBetweenV1AndV2 means that we're reading a legacy V1 file and
+	// should just skip reading the last two fields
+	if numFieldsToSkip < 0 && numFieldsToSkip != -numFieldsAddedToIndexInfoBetweenV1AndV2 {
+		dec.err = fmt.Errorf("numFieldsToSkip was less than zero and not: %d",
+			-numFieldsAddedToIndexInfoBetweenV1AndV2)
+	}
+
+	fmt.Println(numFieldsToSkip)
```

Comment:
Need to remove this yeah?

Reply:
yeah I keep missing these in my refactors...

Reply:
try using `println()` instead of `fmt.Println` when debugging. makes the grep much easier